### ISSUE TITLE
No way to find your deleted object from the info provided in a fine-grained notification?

### ISSUE THREAD

**Issue by anohren** [realm/realm-cocoa#3520]
From what I can see I only get arrays of Ints in the changes object. It works if you just want to use the notifications to update a tableview, but...

## Goals

...Say I have a separate backing array for my data, and I need to remove the respective object from it on a realm delete notification.

## Expected Results

I was thinking something along the lines of a `deletions: [ (index: Int, object: T) ]` in the `changes` object.

**Comment by jpsim** [comment]
You're right, this is a limitation of the current approach. We optimized the current version to be useful in the most common use case people have asked for more fine-grained change information: updating table views.

We intend to continue improving notification functionality over time, including having the old and new state of objects, but that's still early in the design phase.
